iothub/device/src/Authentication/IDeviceIdentity.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using Microsoft.Azure.Devices.Client.Transport;
+
+namespace Microsoft.Azure.Devices.Client
+{
+    /// <summary>
+    /// Interface for device configurations and common attributes.
+    /// </summary>
+    internal interface IDeviceIdentity : IAuthorizationProvider
```

Comment:
Or `IClientIdentity` if you want to disambiguate between devices and modules.

Reply:
I like that